redfish-core/lib/hypervisor_system.hpp

```diff
@@ -350,60 +422,68 @@ inline void
 }
 
 /**
- * @brief Creates a static IPv4 entry
+ * @brief Creates a static IPv4/6 entry
  *
- * @param[in] ifaceId      Id of interface upon which to create the IPv4 entry
- * @param[in] prefixLength IPv4 prefix syntax for the subnet mask
- * @param[in] gateway      IPv4 address of this interfaces gateway
- * @param[in] address      IPv4 address to assign to this interface
+ * @param[in] ifaceId      Id of interface upon which to create the IPv4/6 entry
+ * @param[in] prefixLength IPv4/6 prefix length
+ * @param[in] gateway      IPv4/6 address of this interfaces gateway
+ * @param[in] address      IPv4/6 address to assign to this interface
  * @param[io] asyncResp    Response object that will be returned to client
  *
  * @return None
  */
 inline void
-    createHypervisorIPv4(const std::string& ifaceId, const uint8_t prefixLength,
-                         const std::string& gateway, const std::string& address,
-                         const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+    createHypervisorIP(const std::string& ifaceId, const uint8_t prefixLength,
+                       const std::string& gateway, const std::string& address,
+                       const std::string& protocol,
+                       const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
 {
     crow::connections::systemBus->async_method_call(
-        [asyncResp](const boost::system::error_code ec) {
+        [asyncResp, prefixLength, gateway,
+         address](const boost::system::error_code ec) {
             if (ec)
             {
                 BMCWEB_LOG_DEBUG
-                    << "createHypervisorIPv4 failed: ec: " << ec.message()
+                    << "createHypervisorIP failed: ec: " << ec.message()
                     << " ec.value= " << ec.value();
-                if ((ec == boost::system::errc::invalid_argument) ||
-                    (ec == boost::system::errc::argument_list_too_long))
+                if (ec == boost::system::errc::invalid_argument)
+                {
+                    messages::actionParameterUnknown(asyncResp->res, "Create",
+                                                     "IPv4StaticAddresses");
+                }
+                else if (ec == boost::system::errc::argument_list_too_long)
                 {
-                    messages::invalidObject(asyncResp->res,
-                                            "IPv4StaticAddresses");
+                    std::string createInp = address + "; " +
```

Comment:
It should be a comma separated string correct? Semicolon is given

Reply:
its just a string that will be displayed as one of the arguments in the error message. Do you still think it should be comma separated?

Reply:
Reverting back to InvalidObject error